app/services/utils/current_work_stream_calculator.rb

```diff
@@ -0,0 +1,25 @@
+module Utils
+  class CurrentWorkStreamCalculator
+    attr_reader :work_stream_param, :session_work_stream, :user_competencies
+
+    def initialize(work_stream_param:, session_work_stream:, user_competencies:)
+      @work_stream_param = work_stream_param
+      @session_work_stream = session_work_stream
+      @user_competencies = user_competencies
+    end
+
+    def current_work_stream
+      if @work_stream_param.present?
```

Comment:
As you're listing these attr in the attr_reader use "work_stream_param" rather than the instance_variable "@work_stream_param"